### DISCUSSION TITLE
Reset 2FA Token

### DISCUSSION THREAD

**Discussion by Belnando** [amidaware/tacticalrmm#721] [Q&A] [answered]
I have used the command line script to reset my 2fa token and was expecting a new QR so that I can scan and create the new token at logon to RMM.  This does not happen, I still get the usual logon, requesting a token rather than showing me the QR.

Secondly when I ran the script it gave me a setup key that I can insert manually... where does that go? 

Thanks for your support.
Belnando

**Comment by dinger1986** [comment]
My suggestion on this would be to discuss on discord so you can share screenshots etc

**Reply by Belnando** [reply]
OK

**Comment by wh1te909** [comment, marked as answer]
assuming you're running `python manage.py reset_2fa <username>` it should print a barcode, see screenshot below

<img width="852" alt="Screen Shot 2021-09-18 at 11 33 53 AM" src="https://user-images.githubusercontent.com/7434746/133905098-63ec1076-4e0f-44bb-8d4b-f4b355b0f65d.png">

the setup key is provided in case the barcode doesn't print for whatever reason (not sure why it's not printing for you, maybe try using a different terminal emulator like putty). You have to enter that setup key in your authenticator app, not sure which app you're using but it should give you the option to enter a key instead of scanning a qr code.

**Reply by golg3** [reply]
Sorry, Bad credentials 400: Bad Request

**Reply by dinger1986** [reply]
Check docs for error 400

**Reply by golg3** [reply]
Can I turn off two-factor authentication?

**Reply by dinger1986** [reply]
No

**Reply by golg3** [reply]
I couldn't find a solution from the 400 error documentation.

**Reply by dinger1986** [reply]
https://docs.tacticalrmm.com/troubleshooting/#cant-login-on-server-after-first-setup

**Comment by Belnando** [comment]
Thanks.. I got it going. I did a clean reinstall for the sake of following
and understanding the steps more precisely.

I understand the process much better now. I will keep an eye on the QR side
of things going forward.

Thanks
Belnando

**Comment by ccasalicchio** [comment]
I've ran the following:
```
su - tactical
python /rmm/api/tacticalrmm/manage.py reset_2fa xyz
```

and got this error:
```
Traceback (most recent call last):
  File "/rmm/api/tacticalrmm/manage.py", line 10, in main
    from django.core.management import execute_from_command_line
ModuleNotFoundError: No module named 'django'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/rmm/api/tacticalrmm/manage.py", line 21, in <module>
    main()
  File "/rmm/api/tacticalrmm/manage.py", line 12, in main
    raise ImportError(
ImportError: Couldn't import Django. Are you sure it's installed and available on your PYTHONPATH environment variable? Did you forget to activate a virtual environment?

```

**Comment by ccasalicchio** [comment]
nevermind...
had to do first
```
cd /rmm/api/tacticalrmm
source ../env/bin/activate
```

**Reply by dinger1986** [reply]
lol was just about to say!

https://docs.tacticalrmm.com/management_cmds/#reset-a-users-password 

top of page has that info :)

**Comment by lorranerr** [comment]
Excelente, rodei python manage.py reset_2fa admin e consegui redefinir.

**Comment by JackBHz** [comment]
fala pessoal fiz o processo na linha de comando e consegui redefini, porem cada vez que eu leio o qr code da uma informação diferente e o tento colocar a chave manualmente e não passa na tela de autenticação. vcs tem algum leitor de qr especifico?
no mias vlw

**Reply by silversword411** [reply]
I Google translated your question, but please use English

This is a TOTP compatible code, so any authenticator app will work.

Bitwarden, or Google authenticator or Microsoft authenticator
